#### Let us start by importing our dependencies

In [51]:
import pandas as pd
import numpy as np
from config import api_key
import requests
import json
import scipy.stats as stats

In [11]:
# Create a dataframe of all award-winning beers since 2009
csv_data = "beer_winners.csv"

all = pd.read_csv(csv_data)
all.columns = ['medal', 'beer_name','brewery',
                       'city','county','state',
                       'category','year','total_category_entries']
all = all.sort_values('year', ascending=False)
all.reset_index(inplace=True)
all_winner_data = all.drop(columns=['index'],axis=1)
print("This is a dataframe with every award-winning beer since 2009")
all_winner_data.head()

This is a dataframe with every award-winning beer since 2009


medal                beer_name                             brewery  \
0  Bronze                  Eclipse              FiftyFifty Brewing Co.   
1  Silver                   Märzen                 Sudwerk Brewing Co.   
2    Gold              Oktoberfest                         SKA Brewing   
3    Gold  Barrel Aged Japance Off  Denver Beer Co. - Olde Town Arvada   
4    Gold          3 Barrel Circus               River Dog Brewing Co.   

       city     county state                            category  year  \
0   Truckee     Nevada    CA  Wood- and Barrel-Aged Strong Stout  2018   
1     Davis       Yolo    CA          American-Style Amber Lager  2018   
2   Durango   La Plata    CO                  Vienna-Style Lager  2018   
3    Arvada  Jefferson    CO          Wood- and Barrel-Aged Beer  2018   
4  Bluffton       Yell    SC     Wood- and Barrel-Aged Sour Beer  2018   

   total_category_entries  
0                     177  
1                      91  
2                      92  
3                      77  
4                      76

### How many beers brewed in California have won a medal?

In [13]:
ca = all_winner_data[all_winner_data['state'] == 'CA']
ca.reset_index(inplace=True)
ca_number = ca['index'].count()
print(f'There have been {ca_number} award-winning* beers in California in the last decade.')
cal_data = ca.drop(columns=['index'],axis=1)
cal_data = cal_data.sort_values('year', ascending=False)
cal_data.reset_index(inplace=True)
cal_data = cal_data.drop(columns=['index'],axis=1)
cal_data.head()

There have been 553 award-winning* beers in California in the last decade.


medal         beer_name                        brewery       city  \
0  Bronze           Eclipse         FiftyFifty Brewing Co.    Truckee   
1  Bronze         Nighthawk            Enegren Brewing Co.   Moorpark   
2  Silver  Sour Cherry Sour          Hermitage Brewing Co.   San Jose   
3  Bronze    Kumquat Saison  Smog City Brewing Wood Cellar   Torrance   
4  Silver        The Shroud                Bravery Brewing  Lancaster   

        county state                                 category  year  \
0       Nevada    CA       Wood- and Barrel-Aged Strong Stout  2018   
1      Ventura    CA                               Dark Lager  2018   
2       Graham    CA  Fruited Wood- and Barrel-Aged Sour Beer  2018   
3  Los Angeles    CA                 Belgian-Style Fruit Beer  2018   
4  Los Angeles    CA                           Imperial Stout  2018   

   total_category_entries  
0                     177  
1                      62  
2                     102  
3                      65  
4                      86

### How many beers brewed in San Diego have won a medal?

In [14]:
temp = cal_data[cal_data['county'] == 'San Diego']
temp.reset_index(inplace=True)
sd_number = temp['index'].count()
print(f'There have been {sd_number} award-winning* beers in San Diego in the last decade.')
sd_data = temp.drop(columns=['index'], axis=1)
sd_data = sd_data.sort_values('year', ascending=False)
sd_data.reset_index(inplace=True)
sd_data = sd_data.drop(columns=['index'],axis=1)
sd_data.head()

There have been 148 award-winning* beers in San Diego in the last decade.


medal                      beer_name                              brewery  \
0  Silver                           Oats              Pizza Port Solana Beach   
1    Gold   Natural Bridge: Vienna Lager                        Eppig Brewing   
2    Gold                      Liquid AC  Karl Strauss Brewing Co. - La Jolla   
3  Bronze                      Bird Park                  North Park Beer Co.   
4  Bronze  Chocolate Macadamia Nut Stout                     Kilowatt Brewing   

           city     county state                  category  year  \
0  Solana Beach  San Diego    CA             Oatmeal Stout  2018   
1     San Diego  San Diego    CA              Session Beer  2018   
2      La Jolla  San Diego    CA  English-Style Summer Ale  2018   
3     San Diego  San Diego    CA   Bohemian-Style Pilsener  2018   
4     San Diego  San Diego    CA            Chocolate Beer  2018   

   total_category_entries  
0                      59  
1                      50  
2                      61  
3                      84  
4                      59

### In terms of medals won, what are the top 5 breweries in San Diego?

In [16]:
medals_sd = sd_data.groupby(['brewery'])['medal'].count()
top_brew_sd = medals_sd.sort_values(ascending=False)
top_5_sd = top_brew_sd.head()
top_5_sd

brewery
Pizza Port Carlsbad        24
AleSmith Brewing Co.       10
The Lost Abbey              9
Pizza Port Solana Beach     8
Pizza Port Ocean Beach      7
Name: medal, dtype: int64

### In terms of medals won, what are the top 5 breweries in California?

In [49]:
medals_cal = cal_data.groupby(['brewery'])['medal'].count()
top_brew_cal = medals_cal.sort_values(ascending=False)
top_5_cal = top_brew_cal.head()
top_5_cal

brewery
Firestone Walker Brewing Co.    28
Pizza Port Carlsbad             24
Pabst Brewing Co.               12
Pizza Port Ocean Beach          12
Beachwood BBQ & Brewing         11
Name: medal, dtype: int64




### Why is there a discrepancy between the San Diego list and the California list?
#### Let's find out!




In [72]:
pizza_port = all_winner_data[all_winner_data['brewery'] == 'Pizza Port Ocean Beach']
pizza_port = pizza_port.sort_values('year', ascending=False)
pizza_port.reset_index(inplace=True)
pizza_port = pizza_port.drop(columns=['index'],axis=1)
pizza_port

medal                      beer_name                 brewery  \
0     Gold                      Guillaume  Pizza Port Ocean Beach   
1   Bronze                 Bacon and Eggs  Pizza Port Ocean Beach   
2   Bronze                      Guillaume  Pizza Port Ocean Beach   
3   Silver                      Guillaume  Pizza Port Ocean Beach   
4   Bronze                La Flama Dorada  Pizza Port Ocean Beach   
5     Gold                    Beer Hunter  Pizza Port Ocean Beach   
6   Silver                   Rhino Chaser  Pizza Port Ocean Beach   
7   Bronze                  Kung Fu Elvis  Pizza Port Ocean Beach   
8     Gold             Skidmark Brown Ale  Pizza Port Ocean Beach   
9   Silver  Rhino Chaser Imperial Red Ale  Pizza Port Ocean Beach   
10  Bronze                    Z-Man Stout  Pizza Port Ocean Beach   
11  Silver                 Skidmark Brown  Pizza Port Ocean Beach   

           city     county state                        category  year  \
0     San Diego  San Diego    CA                    Session Beer  2017   
1     San Diego  San Diego    CA                     Coffee Beer  2015   
2     San Diego  San Diego    CA                    Session Beer  2015   
3   Ocean Beach    Suffolk    CA                    Session Beer  2014   
4   Ocean Beach    Suffolk    CA            Belgian-Style Tripel  2014   
5   Ocean Beach    Suffolk    CA                    Session Beer  2013   
6   Ocean Beach    Suffolk    CA                Imperial Red Ale  2013   
7   Ocean Beach    Suffolk    CA  American-Style Strong Pale Ale  2013   
8     San Diego  San Diego    CA          English-Style Mild Ale  2011   
9     San Diego  San Diego    CA                Imperial Red Ale  2011   
10    San Diego  San Diego    CA             Foreign-Style Stout  2011   
11    San Diego  San Diego    CA          English-Style Mild Ale  2010   

    total_category_entries  
0                       40  
1                      149  
2                       44  
3                       94  
4                       58  
5                       56  
6                       56  
7                      120  
8                       25  
9                       57  
10                      23  
11                      27

### Now for the big question:
### What are the top 5 breweries in the United States?

In [53]:
medals_us = all_winner_data.groupby(['brewery'])['medal'].count()
top_brew_us = medals_us.sort_values(ascending=False)
top_5_us = top_brew_us.head()
top_5_us

brewery
Firestone Walker Brewing Co.    28
Pizza Port Carlsbad             24
Sun King Brewing Co.            19
Pabst Brewing Co.               19
Chuckanut Brewery               18
Name: medal, dtype: int64

In [ ]:
# ^^^
# Maybe we might want to speak on the fact that
# Firestone is in SLO and Pizza Port Carlsbad is in SD County?

In [62]:
# Create a dataframe?
# Possible visualization ideas?
# Possible state-by-state breakdown of medals? Using a different groupby?
# Even see the mean of all US states medal counts and compare it to CA? <-- I like this idea the best so far, myself

sd_medals = sd_data.groupby(['medal'])['year'].count()
ca_medals = cal_data.groupby(['medal'])['year'].count()
all_medals = all_winner_data.groupby(['medal'])['year'].count()

medals_df = pd.DataFrame({"SD medals won": sd_medals,
                         "CA medals won": ca_medals,
                         "All US won": all_medals})

medals_df['SD expected'] = [49, 49, 49]
medals_df['CA expected'] = [184, 184, 184]
medals_df['US expected'] = [882, 882, 882]

In [61]:
sd_df = medals_df[['SD medals won', 'SD expected']]
sd_df

SD medals won  SD expected
medal                             
Bronze             49           49
Gold               58           49
Silver             41           49

In [65]:
crit_sd = stats.chi2.ppf(q = 0.95, df = 2)
print(f'The critical value for San Diego medal frequency is {crit_sd}')
stats.chisquare(sd_df['SD medals won'], sd_df['SD expected'])

The critical value for San Diego medal frequency is 5.991464547107979


Power_divergenceResult(statistic=2.9591836734693877, pvalue=0.2277306206904969)

### Conclusion
* Since the chi-square value of 2.96 at a confidence level of 95% does not exceed the critical value of 5.99, we conclude that the differences seen in the number of wins per medal are not statistically significant.
* If you were handed a randomly-chosen award-winning beer in San Diego, CA, it is just as likely to be a gold-medal beer as a bronze-medal beer.


In [66]:
ca_df = medals_df[['CA medals won', 'CA expected']]
ca_df

CA medals won  CA expected
medal                             
Bronze            186          184
Gold              185          184
Silver            182          184

In [68]:
crit_ca = stats.chi2.ppf(q = 0.95, df = 2)
print(f'The critical value for San Diego medal frequency is {crit_ca}')
stats.chisquare(ca_df['CA medals won'], ca_df['CA expected'])

The critical value for San Diego medal frequency is 5.991464547107979


Power_divergenceResult(statistic=0.048913043478260865, pvalue=0.9758401158271994)

In [102]:
means_us = all_winner_data.groupby(['category'])['total_category_entries'].mean()
means_us.head(10)

category
Aged Beer                                    33.2
American Style Amber Lager                   45.0
American Style Amber/Red Ale                 78.0
American Style Brown Ale                     43.0
American Style Light Lager                   25.0
American Style Pale Ale                     108.0
American Style Specialty Lager               23.0
American-Belgo-Style Ale                     60.7
American-Style Amber Lager                   58.0
American-Style Amber Lager or Dark Lager    108.0
Name: total_category_entries, dtype: float64

In [101]:
means_cal = cal_data.groupby(['category'])['total_category_entries'].mean()
means_cal.head(10)

category
Aged Beer                                    34.333333
American Style Amber/Red Ale                 78.000000
American Style Brown Ale                     43.000000
American-Belgo-Style Ale                     63.375000
American-Style Amber Lager                   69.500000
American-Style Amber Lager or Dark Lager    108.000000
American-Style Amber/Red Ale                 96.363636
American-Style Black Ale                     63.625000
American-Style Brett Beer                    38.000000
American-Style Brown Ale                     75.142857
Name: total_category_entries, dtype: float64

In [100]:
means_sd = sd_data.groupby(['category'])['total_category_entries'].mean()
means_sd.head(10)

category
American Style Amber/Red Ale       78.000000
American Style Brown Ale           43.000000
American-Belgo-Style Ale           70.333333
American-Style Amber/Red Ale       99.333333
American-Style Black Ale           73.000000
American-Style Brown Ale           69.666667
American-Style India Black Ale     53.000000
American-Style India Pale Ale     235.000000
American-Style Pale Ale           157.500000
American-Style Sour Ale            53.400000
Name: total_category_entries, dtype: float64

In [116]:
# Possibly a line graph? Or scatter-plot?

total_entries_us = all_winner_data.groupby(['year'])['total_category_entries'].count()
print('total US entries')
print(total_entries_us)
total_entries_cal = cal_data.groupby(['year'])['total_category_entries'].count()
print("")
print('total CA entries')
print(total_entries_cal)
total_entries_sd = sd_data.groupby(['year'])['total_category_entries'].count()
print("")
print('total SD entries')
print(total_entries_sd)

total US entries
year
2009    234
2010    236
2011    248
2012    251
2013    249
2014    268
2015    275
2016    286
2017    293
2018    306
Name: total_category_entries, dtype: int64

total CA entries
year
2009    39
2010    52
2011    51
2012    49
2013    52
2014    46
2015    67
2016    68
2017    57
2018    72
Name: total_category_entries, dtype: int64

total SD entries
year
2009    14
2010    15
2011    18
2012    14
2013     8
2014    12
2015    19
2016    18
2017    14
2018    16
Name: total_category_entries, dtype: int64


In [119]:
# List of all award-winning beers in 2018
# Any ides for visualization and/or story-telling?

bweh = sd_data.loc[sd_data['year'] == 2018]
sd_2018 = bweh[['medal', 'beer_name', 'brewery', 'category', 'total_category_entries']]
sd_2018

medal                         beer_name  \
0     Gold                      The Coachman   
1   Silver                              Oats   
2   Silver                   Duck Duck Gooze   
3   Silver                            Redrum   
4   Silver                       Noble Miner   
5   Bronze                         Bird Park   
6   Bronze                          Know Ego   
7   Bronze                Rail Gun Wee Heavy   
8   Silver            Delaminator Doppelbock   
9   Bronze                    Zungen Brecher   
10    Gold               B.Right On Pale Ale   
11    Gold      Natural Bridge: Vienna Lager   
12    Gold                          Rye Dawn   
13  Bronze     Chocolate Macadamia Nut Stout   
14    Gold                         Liquid AC   
15  Silver  Groundswell Piloncillo Brown Ale   

                                brewery                            category  \
0                   Societe Brewing Co.              Session India Pale Ale   
1               Pizza Port Solana Beach                       Oatmeal Stout   
2     Port Brewing Co. / The Lost Abbey     Wood- and Barrel-Aged Sour Beer   
3                 Duck Foot Brewing Co.          Wood- and Barrel-Aged Beer   
4                      Burgeon Beer Co.                        Session Beer   
5                   North Park Beer Co.             Bohemian-Style Pilsener   
6               Groundswell Brewing Co.       Classic Irish-Style Dry Stout   
7          BNS Brewing & Distilling Co.               Old Ale or Strong Ale   
8                   Rip Current Brewery  German-Style Doppelbock or Eisbock   
9                      Home Brewing Co.                     Historical Beer   
10                  Ocean Beach Brewery             American-Style Pale Ale   
11                        Eppig Brewing                        Session Beer   
12               Breakwater Brewing Co.                            Rye Beer   
13                     Kilowatt Brewing                      Chocolate Beer   
14  Karl Strauss Brewing Co. - La Jolla            English-Style Summer Ale   
15              Groundswell Brewing Co.            American-Style Brown Ale   

    total_category_entries  
0                       92  
1                       59  
2                       76  
3                       77  
4                       50  
5                       84  
6                       54  
7                       37  
8                       31  
9                       38  
10                     170  
11                      50  
12                      57  
13                      59  
14                      61  
15                      76

In [17]:
# POSSIBLE DATA VISUALIZATION FOR THIS LIST?
# ***How many award-winning beers does each brewery have?***

# Maybe a pie chart showing the medal breakdown of SD breweries?
# Or a pie chart showing the top 5-7 breweries, with all the others listed as "Other"

# Maybe combine similar names (i.e. "Ballast Point Brewing Co" and "Ballast Point Brewing Co.")
top_brew_sd.head()

brewery
Pizza Port Carlsbad        24
AleSmith Brewing Co.       10
The Lost Abbey              9
Pizza Port Solana Beach     8
Pizza Port Ocean Beach      7
Name: medal, dtype: int64

In [20]:
sd_data.head()

medal                      beer_name                              brewery  \
0  Silver                           Oats              Pizza Port Solana Beach   
1    Gold   Natural Bridge: Vienna Lager                        Eppig Brewing   
2    Gold                      Liquid AC  Karl Strauss Brewing Co. - La Jolla   
3  Bronze                      Bird Park                  North Park Beer Co.   
4  Bronze  Chocolate Macadamia Nut Stout                     Kilowatt Brewing   

           city     county state                  category  year  \
0  Solana Beach  San Diego    CA             Oatmeal Stout  2018   
1     San Diego  San Diego    CA              Session Beer  2018   
2      La Jolla  San Diego    CA  English-Style Summer Ale  2018   
3     San Diego  San Diego    CA   Bohemian-Style Pilsener  2018   
4     San Diego  San Diego    CA            Chocolate Beer  2018   

   total_category_entries  
0                      59  
1                      50  
2                      61  
3                      84  
4                      59

In [39]:
group = sd_data.groupby('category')['total_category_entries']
aa = group.count()
aa[aa > 2]

#bb = sd_data.groupby('category')['total_category_entries'].mean()
#bb
# cat = group.size()
# cat_df = pd.DataFrame(cat)
# aa = cat.loc[cat_df[0] > 2]
# aa

category
American-Belgo-Style Ale              3
American-Style Amber/Red Ale          3
American-Style Brown Ale              3
American-Style Sour Ale               5
American-Style Stout                  3
American-Style Strong Pale Ale        3
Barley Wine-Style Ale                 4
Belgian-Style Abbey Ale               3
Belgian-Style Strong Specialty Ale    3
Classic Irish-Style Dry Stout         4
Coffee Beer                           4
English-Style Mild Ale                3
English-Style Summer Ale              3
Imperial India Pale Ale               3
Imperial Red Ale                      6
Irish-Style Red Ale                   3
Oatmeal Stout                         4
Old Ale or Strong Ale                 3
Other Belgian-Style Ale               3
Scotch Ale                            4
Session Beer                          9
Session India Pale Ale                3
Name: total_category_entries, dtype: int64